backend/app/src/main/java/ch/zhaw/studyflow/controllers/CalendarController.java

```diff
@@ -0,0 +1,252 @@
+            return response;
+        });
+    }
+
+    @Route(path = "getCalendar")
+    @Endpoint(method = HttpMethod.GET)
+    public HttpResponse getCalendar(RequestContext context) {
+        final HttpRequest request = context.getRequest();
+        HttpResponse response = context.getRequest().createResponse();
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            request.getRequestBody()
+                    .flatMap(body -> body.tryRead(Calendar.class))
+                    .flatMap(calendar -> {
+                        Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID).map(Long::valueOf);
+                        if (userId.isPresent()) {
+                            Calendar foundCalendar = calendarManager.read(calendar.getId(), userId.get());
+                            return Optional.of(foundCalendar);
+                        } else {
+                            return Optional.empty();
+                        }
+                    })
+                    .ifPresentOrElse(
+                            calendar -> {
+                                response.setResponseBody(JsonContent.writableOf(calendar));
+                                response.setStatusCode(HttpStatusCode.OK);
+                            },
+                            () -> {
+                                response.setStatusCode(HttpStatusCode.BAD_REQUEST);
+                            }
+                    );
+            return response;
+        });
+    }
+
+    @Route(path = "getCalendars")
+    @Endpoint(method = HttpMethod.GET)
+    public HttpResponse getCalendars(RequestContext context) {
+        final HttpRequest request = context.getRequest();
+        HttpResponse response = context.getRequest().createResponse();
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID).map(Long::valueOf);
+            if (userId.isPresent()) {
+                List<Calendar> calendars = calendarManager.getCalendars();
+                response.setResponseBody(JsonContent.writableOf(calendars))
+                        .setStatusCode(HttpStatusCode.OK);
+            } else {
+                response.setStatusCode(HttpStatusCode.BAD_REQUEST);
+            }
+            return response;
+        });
+    }
+
+    @Route(path = "getAppointments")
+    @Endpoint(method = HttpMethod.GET)
+    public HttpResponse getAppointments(RequestContext context) {
+        final HttpRequest request = context.getRequest();
+        HttpResponse response = context.getRequest().createResponse();
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID).map(Long::valueOf);
+            if (userId.isPresent()) {
+                List<Appointment> appointments = appointmentManager.readAllBy(userId.get(), null, null);
```

Comment:
I think it would be a good idea to set a default date range if no date range is set.